sdk/textanalytics/azure-ai-textanalytics/tests/test_recognize_pii_entities.py

```diff
@@ -87,11 +87,69 @@ def test_all_successful_passing_text_document_input(self, client):
 
     @GlobalTextAnalyticsAccountPreparer()
     @TextAnalyticsClientPreparer()
-    def test_length_with_emoji(self, client):
+    def test_emoji(self, client):
         result = client.recognize_pii_entities(["👩 SSN: 859-98-0987"])
         self.assertEqual(result[0].entities[0].offset, 7)
         self.assertEqual(result[0].entities[0].length, 11)
 
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_emoji_with_skin_tone_modifier(self, client):
+        result = client.recognize_pii_entities(["👩🏻 SSN: 859-98-0987"])
+        self.assertEqual(result[0].entities[0].offset, 8)
+        self.assertEqual(result[0].entities[0].length, 11)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_emoji_family(self, client):
+        result = client.recognize_pii_entities(["👩‍👩‍👧‍👧 SSN: 859-98-0987"])
+        self.assertEqual(result[0].entities[0].offset, 13)
+        self.assertEqual(result[0].entities[0].length, 11)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_emoji_family_with_skin_tone_modifier(self, client):
+        result = client.recognize_pii_entities(["👩🏻‍👩🏽‍👧🏾‍👦🏿 SSN: 859-98-0987"])
+        self.assertEqual(result[0].entities[0].offset, 17)
+        self.assertEqual(result[0].entities[0].length, 11)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_diacritics_nfc(self, client):
+        result = client.recognize_pii_entities(["año SSN: 859-98-0987"])
+        self.assertEqual(result[0].entities[0].offset, 9)
+        self.assertEqual(result[0].entities[0].length, 11)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_diacritics_nfd(self, client):
+        result = client.recognize_pii_entities(["año SSN: 859-98-0987"])
+        self.assertEqual(result[0].entities[0].offset, 10)
+        self.assertEqual(result[0].entities[0].length, 11)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_korean_nfc(self, client):
+        result = client.recognize_pii_entities(["아가 SSN: 859-98-0987"])
+        self.assertEqual(result[0].entities[0].offset, 8)
+        self.assertEqual(result[0].entities[0].length, 11)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_korean_nfd(self, client):
+        result = client.recognize_pii_entities(["아가 SSN: 859-98-0987"])
+        self.assertEqual(result[0].entities[0].offset, 8)
+        self.assertEqual(result[0].entities[0].length, 11)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer()
+    def test_zalgo_text(self, client):
+        result = client.recognize_pii_entities(["ơ̵̧̧̢̳̘̘͕͔͕̭̟̙͎͈̞͔̈̇̒̃͋̇̅͛̋͛̎́͑̄̐̂̎͗͝m̵͍͉̗̄̏͌̂̑̽̕͝͠g̵̢̡̢̡̨̡̧̛͉̞̯̠̤̣͕̟̫̫̼̰͓̦͖̣̣͎̋͒̈́̓̒̈̍̌̓̅͑̒̓̅̅͒̿̏́͗̀̇͛̏̀̈́̀̊̾̀̔͜͠͝ͅ SSN: 859-98-0987"])
```

Comment:
interesting, what language is this?

Reply:
i'm not sure, it's called zalta text, it's part of the [list](https://teams.microsoft.com/l/message/19:d6ff4003f5c848a2a2d6dcdfc0ebd497@thread.skype/1582746506991?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1582746506991&teamName=Azure%20SDK&channelName=Service%20-%20Text%20Analytics&createdTime=1582746506991) that brian compiled of scenarios to test for encoding